internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -162,6 +163,19 @@ func (av agentVersion) String() string {
 func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string, action *fleetapi.ActionUpgrade, det *details.Details, skipVerifyOverride bool, skipDefaultPgp bool, pgpBytes ...string) (_ reexec.ShutdownCallbackFn, err error) {
 	u.log.Infow("Upgrading agent", "version", version, "source_uri", sourceURI)
 
+	currentVersion := agentVersion{
+		version:  release.Version(),
+		snapshot: release.Snapshot(),
+		hash:     release.Commit(),
+	}
+
+	// check version before download
+	same, _ := isSameVersion(u.log, currentVersion, packageMetadata{}, version)
+	if same {
+		u.log.Warnf("Upgrade action skipped: %v", ErrUpgradeSameVersion)
```

Comment:
Please log the version we tried to upgrade to, something like 
```suggestion
		u.log.Warnf("Upgrade action skipped because agent is already at version %s", currentVersion)
```

Same at line 226